### PR TITLE
Add unique constraint for add-on version string

### LINKED ISSUE #14332: Resolve duplicate versions for an add-on to allow the version string to be unique per add-on
To date, there are about 2800 submissions with a duplicate version string (the version number). 16 of these affected webextensions [1].

We should resolve these duplicates so we can make the version string unique (per add-on).

[1] Query for reference:
```
select
  v.addon_id, v.version, count(v.id)
  from versions v
  where v.id in (
    select f.version_id
    from files f
    where f.is_signed = 1 and f.is_webextension = 1)
  group by v.addon_id, v.version having count(v.id) > 1;
```

### LINKED ISSUE #16901: Add unique database constraint to prevent duplicate versions for a given add-on
`version` string should be unique per add-on. We enforce that in code, but we should have a database constraint, when the [task to remove existing duplicates](https://github.com/mozilla/addons-server/issues/14332) has been run in prod.

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #16901.

This depends on some preparation work (see issue), so it can't be merged right now.

### CODE REVIEW

File: src/olympia/versions/models.py [hunk 1]
```diff
@@ -215,6 +215,9 @@ class Meta(ModelBase.Meta):
             models.Index(fields=('addon',), name='addon_id'),
             models.Index(fields=('license',), name='license_id'),
         ]
+        constraints = [
+            models.UniqueConstraint(fields=('addon', 'version'), name='version_id'),
```
Comment (eviljeff) [inline src/olympia/versions/models.py:219]: I know this is WIP, but can we have a more descriptive name for this constraint? - it's got nothing to do with a version_id field. 

(the indexes above are also poorly named, but they are least match the column's name - i.e the index for `addon` is actually for `addon_id` in the database table, so the name matches)

Comment (wagnerand) [reply src/olympia/versions/models.py:219]: What do you suggest? I was looking at several other constraints defined in the code and they had similar or even "worse" naming (in terms of meaningfulness), so I went with something that aligns with what's being used in other places.

Comment (diox) [reply src/olympia/versions/models.py:219]: Note that the name needs to be unique. I would suggest trying to follow the name `unique_together` would generate, which is `versions_addon_id_version_5a2e75b6_uniq`.

Comment (wagnerand) [reply src/olympia/versions/models.py]: dc9177812d

**Review by diox — APPROVED** [review]
Should not be merged before https://github.com/mozilla/addons-server/issues/16800 is fired in all environments (so it will likely land in **2021.04.15** and not before)

**Comment by wagnerand** [comment]
We decided not to create a new task in #14332 and instead run it manually on Monday, so this can and shall land for 2021.04.15.

**Comment by diox** [comment]
We've added the fix command to [this week's push notes](https://github.com/mozilla/addons/blob/main/releases/2021/04/15.md) in the "before we start" section and notified ops about it.

**Pull request merged**